### PR TITLE
Rename loaders (binary, library)

### PR DESCRIPTION
**Pull request by ipetr0v**
This change resolves single app server name ambiguity by introducing `oak_loader` library and `oak_runner` binary.

Fixes https://github.com/project-oak/oak/issues/569

### Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [ ] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [x] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: oak/server/asylo/BUILD [hunk 1]
```diff
@@ -170,8 +170,8 @@ enclave_loader(
 )
 
 enclave_loader(
-    name = "oak_app_loader",
-    srcs = ["asylo_oak_app_loader_main.cc"],
+    name = "oak_runner",
```
Comment (tiziano88) [inline oak/server/asylo/BUILD]: What made you decide against `oak_runner_asylo` and `oak_runner_dev` in the end, which is what #569 references? Not necessarily against it either way, but I'm just curious why the change.

In any case, I think the binary name should match the source file name, or at least follow a similar structure.

Comment (ipetr0v) [reply oak/server/asylo/BUILD]: I just thought that the `asylo` is already implied by the directory.
Changed to `asylo_oak_runner`.
Setting `asylo_` at the beginning since all our files follow this convention.
